include/eve/module/real/core/function/regular/generic/significants.hpp

```diff
@@ -0,0 +1,38 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2020 Joel FALCOU
+  Copyright 2020 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/function/exp10.hpp>
+#include <eve/function/convert.hpp>
+#include <eve/function/abs.hpp>
+#include <eve/function/ceil.hpp>
+#include <eve/function/nearest.hpp>
+#include <eve/function/log10.hpp>
+#include <eve/detail/skeleton_calls.hpp>
+#include <eve/concept/value.hpp>
+#include <eve/detail/apply_over.hpp>
+
+namespace eve::detail
+{
+  template<floating_real_value T, integral_value U>
+  EVE_FORCEINLINE T significants_(EVE_SUPPORTS(cpu_)
+                               , T const &a
+                               , U const &n) noexcept
+  {
+    if constexpr(has_native_abi_v<T> && has_native_abi_v<U>)
+    {
+      using elt_t =  element_type_t<T>;
+      auto factor = exp10(convert(n, as<elt_t>())- ceil(log10(eve::abs(a))));
+      return nearest(a * factor) / factor;
+    }
+    else  return apply_over(significants, a, n);
+  }
```

Comment:
Questions: 
1) why do you need a check for `has_native_abi`? Should just work, no?
2) Should you, maybe, have a requirement that T and U have the same cardinality?

Reply:
the problem is:
1) it is better to split the entries to natives has soon as possible, as it reduces the splits numbers as apply over does so
2) T and U can be wide or scalar: apply_over take also care of that 
I hope this is a correct answer

Reply:
I didn't think about scalar - that makes sense.

In general - if you want to see the codegen, you can disassemble any cpp file, similar to compiler explorer.

```
clang++ -std=c++20 --stdlib=libc++ -O3 -DEVE_DISABLE_ASSERTS -march=native -I  ../../eve/include -I ../src -S -mllvm --x86-asm-syntax=intel disassemble.cc
```

Is the command I use

Reply:
thank you, I will try